process/wsclient/client.go

```diff
@@ -27,108 +34,86 @@ var (
 )
 
 type client struct {
-	urlReceive string
-	actions    map[data.OperationType]func(marshalledData []byte) error
+	urlReceive               string
+	closeActions             func() error
+	actions                  map[data.OperationType]func(marshalledData []byte) error
+	uint64ByteSliceConverter websocketOutportDriver.Uint64ByteSliceConverter
+	wsConnection             wsConn
 }
 
-func NewWebSocketClient(urlReceive string, actions map[data.OperationType]func(marshalledData []byte) error) (*client, error) {
-	urlReceiveData := url.URL{Scheme: "ws", Host: fmt.Sprintf(urlReceive), Path: "/operations"}
+// New will create a new instance of websocket client
+func New(
+	urlReceive string,
+	operationsHandler operationsHandler,
+) (*client, error) {
+	urlReceiveData := url.URL{Scheme: "ws", Host: urlReceive, Path: data.WSRoute}
 
 	return &client{
-		actions:    actions,
-		urlReceive: urlReceiveData.String(),
+		actions:                  operationsHandler.GetOperationsMap(),
+		closeActions:             operationsHandler.Close,
+		urlReceive:               urlReceiveData.String(),
+		uint64ByteSliceConverter: uint64ByteSlice.NewBigEndianConverter(),
 	}, nil
 }
 
+// Start will initialize the connection to the server and start to listen for messages
 func (c *client) Start() {
-	interrupt := make(chan os.Signal, 1)
-	signal.Notify(interrupt, os.Interrupt)
-
 	log.Info("connecting to", "url", c.urlReceive)
 
-	var wsConnection *websocket.Conn
-	done := make(chan struct{})
-	go func() {
-		defer close(done)
-		for {
-			var err error
-			wsConnection, err = c.openConnection()
-			if err != nil {
-				log.Warn(fmt.Sprintf("websocket error, retrying in %v...", retryDuration), "error", err.Error())
-				time.Sleep(retryDuration)
-				continue
-			}
-
-			c.listeningOnWebSocket(wsConnection)
+	for {
+		err := c.openConnection()
+		if err != nil {
+			log.Warn(fmt.Sprintf("c.openConnection(), retrying in %v...", retryDuration), "error", err.Error())
 			time.Sleep(retryDuration)
+			continue
 		}
-	}()
-
-	timer := time.NewTimer(time.Second)
-	defer timer.Stop()
 
-	for {
-		select {
-		case <-done:
-			return
-		case <-timer.C:
-		case <-interrupt:
-			log.Info("interrupt")
-			if wsConnection == nil {
-				return
-			}
-
-			// Cleanly close the connection by sending a close message and then
-			// waiting (with timeout) for the server to close the connection.
-			err := wsConnection.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
-			if err != nil {
-				log.Error("write close", "error", err)
-				return
-			}
-			select {
-			case <-done:
-			case <-time.After(time.Second):
-			}
+		closed := c.listeningOnWebSocket()
+		if closed {
```

Comment:
so this process will immediately close if the node process closes but when it is first started, it will wait until the node process boots up. Is this correct?

Reply:
If it is first started will close immediately, and it will wait on the close when writing something in the Elasticsearch database